sci-physics/rivet/rivet-2.6.0.ebuild

```diff
@@ -0,0 +1,79 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python2_7 )
+
+inherit autotools bash-completion-r1 python-single-r1
+
+DESCRIPTION="Toolkit for validation of Monte Carlo HEP event generators"
+HOMEPAGE="http://rivet.hepforge.org/"
+
+SRC_URI="http://www.hepforge.org/archive/${PN}/${P^}.tar.bz2
+	doc? ( https://rivet.hepforge.org/trac/export/8a05acecd26c18f368f4b748da2d5aa2db46be6c/doc/refs.bib )"
+LICENSE="GPL-2"
+
+SLOT="0"
+KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux"
+IUSE="doc python static-libs"
+REQUIRED_USE="python? ( ${PYTHON_REQUIRED_USE} )"
+
+RDEPEND="
+	dev-libs/boost:=
+	sci-libs/gsl:=
+	sci-physics/fastjet[plugins]
+	sci-physics/hepmc
+	>=sci-physics/yoda-1.5.0[python]
+	python? ( ${PYTHON_DEPS} )"
+DEPEND="${RDEPEND}
+	doc? ( app-doc/doxygen[latex,dot] )
+	python? ( dev-python/cython[${PYTHON_USEDEP}] )"
+
+S="${WORKDIR}/${P^}"
+
+pkg_setup() {
+	use python && python-single-r1_pkg_setup
+}
+
+src_unpack() {
+	unpack "${P^}.tar.bz2"
+
+	if use doc; then
+		# refs.bib is missing in tarball (reported upstream)
+		cp "${DISTDIR}"/refs.bib "${S}"/doc || die
+	fi
+}
+
+src_prepare() {
+	default
+
+	# Install rivet-manual.pdf to docdir intead of pkgdatadir
+	sed -i '/pkgdata_DATA = $(DOCS)/s/pkgdata/doc/' doc/Makefile.am || die
+	eautoreconf
+}
+
+src_configure() {
+	append-cxxflags "-std=c++1z"
```

Comment:
We cannot force flags like this, unless they are required by the package itself (i.e. the package is C++17 only). What if my compiler does not support C++17? Note only GCC 7.3.0 and above have full support for C++17, even though the flag may exist in earlier versions.

Reply:
The problem is that without the C++11 support the package can't compile because it uses C++11 standard in the code, so I thought that the only way to fix it was forcing the flag, am I wrong?

Reply:
Ah, you added a flag that forces C++17, not C++11. It's ok to add a flag for C++11 in that case.

Reply:
With the C++11 flags it produced some warnings during compilation but it built, using the C++17 the warning gone away so I thought It could be reasonable to force it. Anyway I got the point about the compatibility and I corrected It.